### DISCUSSION TITLE
Deploy without overwriting.

### DISCUSSION THREAD

**Discussion by StuComCal** [controlm/ctm-python-client#64] [Q&A] [answered]
Hi, im having an issue when I deploy using the tool it overwrites everything in the folder if i just have one new job. I understand that when we deploy we need to deploy the whole control m folder. But is there a function to populate a work flow with a folder already on the server so I can just add the new job and then redeploy.

The use case is like this cause we both have control-m deployments through the client application as well as the code automatic deployment and was hoping not to interfere.

hope you have a good day.

**Comment by nathanw9722** [comment, marked as answer]
Hi @StuComCal, you do not have to deploy the entire folder. You can use the pathElement to deploy a specific job. Here's a link to the docs: [pathElement](https://documents.bmc.com/supportu/API/Monthly/en-US/Documentation/API_CodeRef_JobProperties.htm#PathElement).

**Reply by StuComCal** [reply]
Thank you, I gave it a try with the below json and got the below error so not quite sure where I'm going wrong.

Error: "message" : "PathElement is an unknown keyword therefore it is assumed to be an object, but it has no object syntax"

{
    "SALARIES_JANUARY_2022": {
        "Type": "Job:Command",
        "PathElement": {
            "Folder": "001deploytest"
        },
        "Command": "echo my 231st job"
    }
}

**Reply by jfink5** [reply]
Hi @StuComCal, I think PathElement was added in 9.0.21.  Try upgrading to latest version of API.

**Reply by StuComCal** [reply]
Thanks @jfink5 unfortunately it its a company server I will speak to the team to see if we can upgrade the API,  do appreciate all the answers and repsonses though.

**Comment by jfink5** [comment]
Upgrading the AAPI has no impact on the rest of Control-M.

________________________________
Jon Fink
Principal Solution Engineer
DBA PreSales
Mobile

+1 239.821.8675
USA Florida.- Office at Home

[BMC—Run and Reinvent]<http://www.bmc.com/>
[BMC Communities]<https://communities.bmc.com/>
[Twitter]<https://twitter.com/bmcsoftware>
[Facebook]<https://www.facebook.com/bmcsoftware>
[YouTube]<http://www.youtube.com/user/BMCSoftwareBMCtv>
[LinkedIn]<http://www.linkedin.com/company/bmc-software>



From: StuComCal ***@***.***>
Sent: Friday, February 23, 2024 2:22 AM
To: controlm/ctm-python-client ***@***.***>
Cc: Fink, Jon ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [controlm/ctm-python-client] Deploy without overwriting. (Discussion #64)


Thanks @jfink5<https://urldefense.com/v3/__https:/github.com/jfink5__;!!PEZBYkTc!bnJhfCKPysNZEqF230azvQnlfV5_qMHVJnXYrhI36vAq6GOsSdhrzOa0IuVlXliv774e7V_b2NQPl_Lk-Z_0ZoM$> unfortunately it its a company server I will speak to the team to see if we can upgrade the API, do appreciate all the answers and repsonses though.
